pkg/infrared/infrared.go

```diff
@@ -88,31 +80,30 @@ func (ir *Infrared) ListenAndServe() error {
 	}
 
 	sgInChan := make(chan ServerRequest)
-	for _, l := range ir.listeners {
-		go func(l net.Listener) {
-			for {
-				c, err := l.Accept()
-				if err != nil {
-					log.Println(err)
-					continue
-				}
+	go func() {
+		for {
+			c, err := ir.l.Accept()
+			if err != nil {
+				// TODO: Handle Listener closed
+				log.Println(err)
+				continue
+			}
 
-				go func(c net.Conn) {
-					conn := newConn(c)
-					defer func() {
-						conn.ForceClose()
-						connPool.Put(conn)
-					}()
+			go func(c net.Conn) {
+				conn := newConn(c)
+				defer func() {
+					conn.ForceClose()
+					connPool.Put(conn)
+				}()
 
-					conn.srvReqChan = sgInChan
+				conn.srvReqChan = sgInChan
 
-					if err := ir.handleConn(conn); err != nil {
-						log.Println(err)
-					}
-				}(c)
-			}
-		}(l)
-	}
+				if err := ir.handleConn(conn); err != nil {
+					log.Println(err)
+				}
+			}(c)
```

Comment:
While we're at it, perhaps we should create a separate function for this, so that we can just do here `go functionName(conn, sgInChan)` 

Just to make the method look simpler,

Reply:
I agree. Should be fixed now